modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/LogsIndexModeCustomSettingsIT.java

```diff
@@ -96,10 +96,15 @@ public void testOverrideIndexSorting() throws IOException {
         assertThat(type, equalTo("date"));
     }
 
-    public void testConfigureStoredSource() throws IOException {
+    public void testConfigureStoredSourceBeforeIndexCreation() throws IOException {
```

Comment:
I had to change this test for the following reason. Here we are removing the `cluster.logsdb.enabled` setting from `StackTemplateRegistry`. This means that when templates are composed using component templates the `index.mode` is not set because StackTemplateRegistry does not inject "logsdb" anymore. It is only after, before the index is created that the failure happens. At that point the `IndexSettingProvider` injects "logsdb" and a conflict is detected by validation while trying to use "logsdb" with store source.

I will add a check for the template to have `index.mode` null soon after `putComponentTemplate`

Reply:
This change in the test makes sense.